src/renderer/widgets/WalletDetails/model/rename-wallet-model.ts

```diff
@@ -0,0 +1,101 @@
+import { attach, combine, createApi, createStore, sample } from 'effector';
+import { createForm } from 'effector-forms';
+import { not } from 'patronum';
+
+import { Wallet } from '@shared/core';
+import { walletModel } from '@entities/wallet';
+
+export type Callbacks = {
+  onSubmit: () => void;
+};
+
+const $callbacks = createStore<Callbacks | null>(null);
+const callbacksApi = createApi($callbacks, {
+  callbacksChanged: (state, props: Callbacks) => ({ ...state, ...props }),
+});
+
+const $walletToEdit = createStore<Wallet | null>(null);
+const walletApi = createApi($walletToEdit, {
+  formInitiated: (state, props: Wallet) => ({ ...state, ...props }),
+});
+
+const $walletForm = createForm({
+  fields: {
+    name: {
+      init: '',
+      rules: [
+        { name: 'required', errorText: 'walletDetails.common.nameRequiredError', validator: Boolean },
+        {
+          name: 'exist',
+          errorText: 'walletDetails.common.nameExistsError',
+          source: combine({
+            walletToEdit: $walletToEdit,
+            wallets: walletModel.$wallets,
+          }),
+          validator: validateNameExist,
+        },
+      ],
+    },
+  },
+  validateOn: ['change', 'submit'],
```

Comment:
`submit` is enough 😉
```suggestion
  validateOn: ['submit'],
```